### PR TITLE
Buffer fixes

### PR DESCRIPTION
**Pull request by stebogit**
I was working on solving #900, and I fixed it simply using the new `@turf/projection`, but then find out that seems to fix also #815 and #916. 😃 

Still for some reason the actual size of the buffer is slightly off, i.e. smaller than the expected one.

cc: @sc-keyzo @ArtCodeKarine @stephanbogner

### CODE REVIEW

**Comment by DenisCarriere** [comment]
@stebogit Wow 🎉 Really cool! The `@turf/projection` module is going to be VERY useful! (we all knew that 😉 )

File: packages/turf-buffer/index.js [hunk 1]
```diff
@@ -94,16 +94,24 @@ function buffer(geojson, radius, units, steps) {
...
-    var projected = {
-        type: geometry.type,
-        coordinates: projectCoords(geometry.coordinates, projection)
-    };
+    var projected;
+    var bbox = turfBbox(geojson);
+    var needsTransverseMercator = bbox[1] > 50 && bbox[3] > 50;
+
+    if (needsTransverseMercator) {
+        var projection = defineProjection(geometry);
```
Comment (DenisCarriere) [inline packages/turf-buffer/index.js]: @stebogit Did you want to move over this `defineProjection` to `@turf/projection` (as `define`).

```js
/**
 * Define Transverse Mercator projection
 *
 * @private
 * @param {Geometry|Feature<any>} geojson Base projection on center of GeoJSON
 * @returns {GeoProjection} D3 Geo Transverse Mercator Projection
 */
function defineProjection(geojson) {
    var coords = center(geojson).geometry.coordinates.reverse();
    var rotate = coords.map(function (coord) { return -coord; });
    var projection = d3.geoTransverseMercator()
        .center(coords)
        .rotate(rotate)
        .scale(6373000);

    return projection;
}
```

**Comment by stebogit** [comment]
Ooops... 😨  I was convinced this branch had been merged already... 😅

File: packages/turf-buffer/index.js (hunk 1 above)
Comment (stebogit) [reply packages/turf-buffer/index.js]: @DenisCarriere wouldn't it be even more useful having a `@turf/projection.toTransverse` method for the actual conversion? I mean, why would you define the (Transverse in this case) projection but for converting the coordinates?
 
In that regard, at this point we might need to re-think the structure of the `projection` module to leave it open to accept maybe other projections in the future, if ever needed, like:
- `@turf/projection.toMercator`
- `@turf/projection.fromMercator`
- `@turf/projection.toTransverse`
- `@turf/projection.fromTransverse`
- ...

knowing the base projection for Turf is `WGS84`

**Comment by DenisCarriere** [comment]
Lol it can be, I only looked at it real quick, when I have more time I'll dive into it more, unless you feel it's good to merge.

**Comment by stebogit** [comment]
At this point I guess we could merge since all tests pass and they actually look better (less distortion); we can always reopen any issue if necessary.
May I have the honor? 😃

**Comment by DenisCarriere** [comment]
@stebogit merge away! 👍

**Comment by stebogit** [comment]
Woo-hoo!! :bowtie:

**Comment by ArtCodeKarine** [comment]
Hello @stebogit, 
I try to install your new module with "npm install @turf/turf", but it doesn't work. Your module isn't display in my project.
So is it normal that your package is not in the file : turf/packages/turf/package.json ?
Thank's for your answer.

**Comment by stebogit** [comment]
@ArtCodeKarine what module are you talking about? Do you mean the updated `@turf/buffer` code?
@DenisCarriere is going to publish a new [patch release](https://github.com/Turfjs/turf/issues/900#issuecomment-329179539) (I guess v4.7.4) with those changes as soon as possible, so stay tuned! 😜

**Comment by DenisCarriere** [comment]
Just published a patch release for `@turf/buffer`, try out:

```bash
$ npm install --save @turf/buffer@4.7.4
```

**Comment by ArtCodeKarine** [comment]
@DenisCarriere @stebogit thanks, I will try it :D

**Comment by ArtCodeKarine** [comment]
I think i have to wait the new version of turf because my application is running with nodejs and bower. Bower uses the @turf/turf folder and this doesn't contain the patch version of the buffer.

**Pull request merged**